### DISCUSSION TITLE
x-for template on table get phantom button

### DISCUSSION THREAD

**Discussion by trialanderror** [alpinejs/alpine#3609] [1. Help] [answered]
Hi, i am trying to get a table from json output. via template / x-for

the table has several columns, and in the last column i add a button.

it loads json information corretly...but it adds a button before the rest of the rows.

![button](https://github.com/alpinejs/alpine/assets/230252/1478f7cb-60f1-4f97-bf51-0529622226f2)

Do i have my json file setup wrong ? or my table ? or my json in alpine ?

`            async getOrders() {
                await fetch(this.url).
                then(jsonFileData=>jsonFileData.json()).
                then((jsonBody)=>{
                        this.ordersLoaded=(jsonBody[0].id!=undefined);
                        if(this.ordersLoaded){
                            this.orderRequest = jsonBody;
                        }
                        }).
                        catch(console.log);
                    },`

                    <table id="myTable" class="table-striped table-hover" style="font-variant:small-caps;font-style:normal;color:black;font-size:12px;">
                        <thead>
                            <tr>
                                <th hidden>ID</th>
                                <th>CUSTOMER</th>
                                <th>WORKORDER</th>
                                <th>ORDER DATE</th>
                                <th>VENDOR</th>
                                <th>DESCRIPTION</th>
                                <th>REQUESTED BY</th>
                                <th>EXPECTED ON</th>
                                <th>RECIEVED ON</th>
                                <th>LOCATION</th>
                                <th>STATUS</th>
                            </tr>
                        </thead>
                        <tbody>
                            <template x-if="ordersLoaded" x-for="order in orderRequest" :key="order.id">
                                <tr>
                                    <td x-text="order.id"></td>
                                    <td x-text="order.customer"></td>
                                    <td x-text="order.workorder"></td>
                                    <td x-text="order.orderdate"></td>
                                    <td x-text="order.vendor"></td>
                                    <td x-text="order.description" class="whitespace-nowrap"></td>
                                    <td x-text="order.requestedby"></td>
                                    <td x-text="order.expected"></td>
                                    <td x-text="order.recieved"></td>
                                    <td x-text="order.location" class="whitespace-nowrap"></td>
                                    <td>
                                        <div class="flex gap-4 items-center justify-center">
                                            <button  type="button" @click="selector(order)" class="btn btn-sm btn-outline-info">SELECT</button>
                                        </div> 
                                    </td>
                                </tr>
                            </template>
                        </tbody>      
                    </table>
`                  
`[
    {
        "customer": "ASCO ODESSA",
        "workorder": "",
        "orderdate": "03\/03\/23",
        "vendor": "MUSTANG",
        "description": "BADGER MINING TEMP SENDER",
        "shipmethod": "DROP SHIP",
        "requestedby": "JARED",
        "expected": "03\/07\/23",
        "recieved": "",
        "location": "",
        "status": "5",
        "id": "168"
    },
    {
        "customer": "BLUE SAGE (DROP SHIPPED)",
        "workorder": "",
        "orderdate": "01\/13\/22",
        "vendor": "LBX",
        "description": "BELTS",
        "shipmethod": "GROUND",
        "requestedby": "STACY",
        "expected": "01\/20\/22",
        "recieved": "",
        "location": "",
        "status": "5",
        "id": "171"
    },
    {
        "customer": "BLUE SAGE ",
        "workorder": "",
        "orderdate": "09\/28\/22",
        "vendor": "CAT ",
        "description": "HARNESSES",
        "shipmethod": "CAT",
        "requestedby": "STERLING",
        "expected": "09\/29\/22",
        "recieved": "",
        "location": "",
        "status": "5",
        "id": "174"
    },
    {
        "customer": "BLUE SAGE",
        "workorder": "",
        "orderdate": "02\/09\/23",
        "vendor": "TAG",
        "description": "COUPLER  ",
        "shipmethod": "GROUND",
        "requestedby": "DALLAS",
        "expected": "14-18 WEEKS",
        "recieved": "05\/24\/23",
        "location": "YARD",
        "status": "2",
        "id": "175"
    }
]`

**Comment by ekwoka** [comment, marked as answer]
It's the `x-if` doing it.

It doesn't work like that.

In this case, you should just have `orderRequest` initialized as an empty array, and then fill it once the stuff is loaded. no reason to use `x-if`

**Reply by trialanderror** [reply]
Thank you...i cant believe it was so easy of a solution..i removed the x-if and now looks good
